packages/wrangler/src/pages/deploy.tsx

```diff
@@ -73,38 +74,49 @@ export function Options(yargs: CommonYargsArgv) {
 		});
 }
 
-export const Handler = async ({
-	_,
-	directory,
-	projectName,
-	branch,
-	commitHash,
-	commitMessage,
-	commitDirty,
-	skipCaching,
-	bundle,
-	noBundle,
-	config: wranglerConfig,
-}: PublishArgs) => {
+export const Handler = async (args: PublishArgs) => {
+	let { branch, commitHash, commitMessage, commitDirty } = args;
+
 	// Check for deprecated `wrangler pages publish` command
-	if (_[1] === "publish") {
+	if (args._[1] === "publish") {
 		logger.warn(
 			"`wrangler pages publish` is deprecated and will be removed in the next major version.\nPlease use `wrangler pages deploy` instead, which accepts exactly the same arguments."
 		);
 	}
 
-	if (wranglerConfig) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
 	}
 
+	// TODO Carmen
+	// this reads the config file with `env` set to undefined, which will
+	// return the top-level config. This won't do for `deploy`, because we
+	// need to know which env we're deploying to. For this to work, we
+	const config = readConfig(undefined, args);
+	if (config.configPath !== "" && config.pages_build_output_dir === undefined) {
+		throw new FatalError(
+			"Found `wrangler.toml` configuration file, but missing `pages_build_output_dir` field required by Pages."
+		);
+	}
+
+	const directory = args.directory ?? config.pages_build_output_dir;
 	if (!directory) {
-		throw new FatalError("Must specify a directory.", 1);
+		throw new FatalError(
+			"Must specify a directory of assets to deploy. Please specify the [<directory>] argument in the `deploy` command, or configure `pages_build_output_dir` in your `wrangler.toml` configuration file.",
+			1
+		);
 	}
 
-	const config = getConfigCache<PagesConfigCache>(PAGES_CONFIG_CACHE_FILENAME);
-	const accountId = await requireAuth(config);
+	const configCache = getConfigCache<PagesConfigCache>(
+		PAGES_CONFIG_CACHE_FILENAME
+	);
+	const accountId = await requireAuth(configCache);
 
-	projectName ??= config.project_name;
+	// TODO Carmen let's talk about config field `name` in Pages
```

Comment:
something I meant to raise a while back but forgot. Since `name` is an inheritable config field, in the context of Pages, can this value actually be different per environment? Like, what would that even mean?  If that doesn't make sense for Pages, whould we error if we find diff project `name`s across the diff envs in `wrangler.toml`?

Reply:
Good catch! We should definitely error in this case

Reply:
We should, but this is inherent to how environments will differ between Pages and Workers. Pages environments will target the _same_ project, while Workers environments target _different_ workers